From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander Khabarov <alexander.khabarov@arm.com>
Date: Tue, 24 Oct 2023 14:52:10 +0000
Subject: [PATCH] Use Windows assembly files for BLAKE3 on Windows

Use Windows assembly files instead of Unix ones when running on
Windows.
---
 llvm/BUILD.bazel | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/llvm/BUILD.bazel b/llvm/BUILD.bazel
index a7e9398ea..ef5299138 100644
--- a/llvm/BUILD.bazel
+++ b/llvm/BUILD.bazel
@@ -220,6 +220,12 @@ cc_library(
         "@platforms//cpu:aarch64": [
             "lib/Support/BLAKE3/blake3_neon.c",
         ],
+        "@bazel_tools//src/conditions:windows_x64": [
+            "lib/Support/BLAKE3/blake3_avx2_x86-64_windows_msvc.asm",
+            "lib/Support/BLAKE3/blake3_avx512_x86-64_windows_msvc.asm",
+            "lib/Support/BLAKE3/blake3_sse2_x86-64_windows_msvc.asm",
+            "lib/Support/BLAKE3/blake3_sse41_x86-64_windows_msvc.asm",
+        ],
         "@platforms//cpu:x86_64": [
             "lib/Support/BLAKE3/blake3_avx2_x86-64_unix.S",
             "lib/Support/BLAKE3/blake3_avx512_x86-64_unix.S",
